components/dashboard/src/data/configurations/configuration-queries.ts

```diff
@@ -88,3 +88,35 @@ export const getConfigurationQueryKey = (configurationId: string) => {
 
     return key;
 };
+
+export type CreateConfigurationArgs = {
+    name: string;
+    cloneUrl: string;
+};
+
+export const useCreateConfiguration = () => {
+    const { data: org } = useCurrentOrg();
+
+    return useMutation<Configuration, Error, CreateConfigurationArgs>({
+        mutationFn: async ({ name, cloneUrl }) => {
+            if (!org) {
+                throw new Error("No org currently selected");
+            }
+
+            // TODO: Should we push this into the api?
+            // ensure a .git suffix
+            const normalizedCloneURL = cloneUrl.endsWith(".git") ? cloneUrl : `${cloneUrl}.git`;
+
+            const response = await configurationClient.createConfiguration({
```

Comment:
Do we need to invalidate the list query when we create a config?

Reply:
Yah, forgot this. I'll also pre-emptively populate the single config query cache so it doesn't have to reload it and wait when we navigate to the settings page on create.